pkg/credentialprovider/config.go

```diff
@@ -30,6 +31,10 @@ import (
 	"k8s.io/klog"
 )
 
+const (
+	MaxReadLength = 1 << 20
```

Comment:
@smarterclayton, these are used to read node-wide credentials as well, right? Is 1MB a reasonable upper bound for that?

Reply:
These functions in the credential provider are used to fetch Docker credential files and data from on-host metadata services. From my testing the contents of these files are typically under 10K, 1MB should be well in excess of even the worst-case credential response. I agree with other reviewers though that we should have better reporting behaviour in the case of an error here.

Reply:
Mike, thanks for the confirmation.
I added some checking on whether the read limit is reached.